services/src/modules/authentication/jwt-decoder-plugin.ts

```diff
@@ -0,0 +1,43 @@
+import * as http from 'http';
+import { decode } from 'jsonwebtoken';
+import { FastifyRequest, FastifyInstance } from 'fastify';
+import * as fp from 'fastify-plugin';
+import logger from '../logger';
+
+type DecodedJWT = {
+  header: Record<string, unknown>;
+  payload: Record<string, unknown>;
+  signature: string;
+};
+
+declare module 'fastify' {
+  interface FastifyRequest<
+    HttpRequest = http.IncomingMessage,
+    Query = DefaultQuery,
+    Params = DefaultParams,
+    Headers = DefaultHeaders,
+    Body = DefaultBody
+  > {
+    decodeJWT(): Promise<void>;
+    decodedJWT?: DecodedJWT | null;
+  }
+}
+
+async function decodeRequestJwt(this: FastifyRequest) {
+  try {
+    const authHeader = this.headers.authorization;
+    if (!authHeader) return;
+    const authToken = String(authHeader).split(' ')[1];
+    const maybeDecodedJWT = (await decode(authToken, { complete: true, json: true })) as DecodedJWT | null;
```

Comment:
```suggestion
    const maybeDecodedJWT = (await decode(authToken, { complete: true, json: true })) as DecodedJWT || null;
```

Reply:
`decode` [returns](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/jsonwebtoken/index.d.ts#L209) `null | { [key: string]: any }`. I just wanted to make the second part more detailed. The single pipe is here intentionally.